javaslang-benchmark/src/test/java/javaslang/MemoryUsage.java

```diff
@@ -1,37 +1,62 @@
 package javaslang;
 
-import javaslang.collection.CharSeq;
-import javaslang.collection.LinkedHashMultimap;
-import javaslang.collection.List;
-import javaslang.collection.Multimap;
+import javaslang.collection.*;
+
+import java.text.DecimalFormat;
+import java.util.regex.Pattern;
 
 import static com.carrotsearch.sizeof.RamUsageEstimator.humanReadableUnits;
 import static com.carrotsearch.sizeof.RamUsageEstimator.sizeOf;
-import static javaslang.BenchmarkPerformanceReporter.DECIMAL_FORMAT;
+import static java.lang.Math.max;
 
 public class MemoryUsage {
-    private static Multimap<Integer, String> memoryUsages = LinkedHashMultimap.withSeq().empty(); // if forked, this will be reset every time
+    private static final DecimalFormat FORMAT = new DecimalFormat("#,##0");
+    private static Map<Integer, LinkedHashSet<Tuple3<CharSeq, CharSeq, CharSeq>>> memoryUsages = TreeMap.empty(); // if forked, this will be reset every time
```

Comment:
@ruslansennov, `Set` is an interface, there is no "default set", returning `HashSet` would be as valid as a `LinkedHashSet` or even `TreeSet`.

But I'm sure you know that, not sure where your hostility originates from.

If you find it offensive that I recommend modifications to your code, no problem I can circumvent it by simulating a multimap also (as done currently).